js/ace.js

```diff
@@ -0,0 +1,1034 @@
+        //         }
+        //     ]
+        //
+        const result = {
+            'info': response,
+        };
+        for (let i = 0; i < response.length; i++) {
+            const balance = response[i];
+            const currencyId = this.safeString (balance, 'currencyName');
+            const code = this.safeCurrencyCode (currencyId);
+            const amount = this.safeString (balance, 'amount');
+            const available = this.safeString (balance, 'cashAmount');
+            const account = {
+                'free': available,
+                'total': amount,
+            };
+            result[code] = account;
+        }
+        return this.safeBalance (result);
+    }
+
+    async fetchBalance (params = {}) {
+        /**
+         * @method
+         * @name ace#fetchBalance
+         * @description query for balance and get the amount of funds available for trading or funds locked in orders
+         * @url https://github.com/ace-exchange/ace-official-api-docs/blob/master/api_v2.md#open-api---account-balance
+         * @param {object} params extra parameters specific to the ace api endpoint
+         * @returns {object} a [balance structure]{@link https://docs.ccxt.com/en/latest/manual.html?#balance-structure}
+         */
+        await this.loadMarkets ();
+        const response = await this.privatePostV2CoinCustomerAccount (params);
+        const balances = this.safeValue (response, 'attachment', []);
+        //
+        //     {
+        //         "attachment":[
+        //             {
+        //                 "currencyId": 4,
+        //                 "amount": 6.896,
+        //                 "cashAmount": 6.3855,
+        //                 "uid": 123,
+        //                 "currencyName": "BTC"
+        //             }
+        //         ],
+        //         message: null,
+        //         parameters: null,
+        //         status: '200'
+        //     }
+        //
+        return this.parseBalance (balances);
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = '/' + this.implodeParams (path, params);
+        const query = this.omit (params, this.extractParams (path));
+        if (headers === undefined) {
+            headers = {};
+        }
+        if (api === 'private') {
+            this.checkRequiredCredentials ();
+            const nonce = this.milliseconds ();
+            let auth = 'ACE_SIGN' + this.secret;
+            const data = this.extend ({
+                'apiKey': this.apiKey,
+                'timeStamp': nonce,
+            }, params);
+            const dataKeys = Object.keys (data);
+            const sortedDataKeys = this.sortBy (dataKeys, 0);
+            for (let i = 0; i < sortedDataKeys.length; i++) {
+                auth += data[sortedDataKeys[i]];
+            }
+            const signature = this.hash (auth, 'sha256', 'hex');
```

Comment:
also, I think we're missing `this.encode(auth)` here otherwise python would break